cmd/juju/cloud/listcredentials.go

```diff
@@ -246,11 +254,23 @@ func (c *listCredentialsCommand) Run(ctxt *cmd.Context) error {
 }
 
 func (c *listCredentialsCommand) remoteCredentials(ctxt *cmd.Context) (map[string]CloudCredential, error) {
-	controllerName, err := c.ControllerNameFromArg()
-	if err != nil && errors.Cause(err) != modelcmd.ErrNoControllersDefined {
-		return nil, errors.Trace(err)
+	if c.ControllerName == "" {
+		// The user may have specified the controller via a --controller option.
+		// If not, let's see if there is a current controller that can be detected.
+		var err error
+		cloudMsg := ""
+		if c.cloudName != "" {
+			cloudMsg = fmt.Sprintf("for cloud %q ", c.cloudName)
+		}
+		c.ControllerName, err = c.MaybePromptCurrentController(ctxt, fmt.Sprintf("list credentials %vfrom", cloudMsg))
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+	if c.ControllerName == "" {
+		return nil, errors.New("Not listing credentials from a controller: no controller specified.")
```

Comment:
can we make the messages and behaviour consistent with list clouds?

Reply:
Hmm, it kind of is ... In 'list-clouds' we specify 
```
No controllers were specified.
You can bootstrap a new controller using one of these clouds:...
```
which is specific to 'clouds' command. So each command has a consistent message to same similar thing - 'no controller specified. not listing/showing blah'